packages/runner/cypress/support/helpers.js

```diff
@@ -219,26 +220,28 @@ function createCypress (defaultOptions = {}) {
         .callsFake(() => {
           autCypress = win.Cypress
 
-          cy.stub(autCypress, 'onSpecWindow').snapshot(enableStubSnapshots).log(false).callsFake((specWindow) => {
-            autCypress.onSpecWindow.restore()
+          if (opts.stubOnSpecWindow) {
```

Comment:
This change is required so that I can actually test clicking a spec from the inline spec list correctly changes the hash and executed the selected spec.

The code for that test is here: https://github.com/cypress-io/cypress/pull/17069/files#diff-03b063928857676b99b170ce6e1787e9fd4bf74acb8c6d06b711abe7adac9378